WordPress/Classes/ViewRelated/Post/NewPostPreviewViewController.swift

```diff
@@ -0,0 +1,252 @@
+        navigationItem.leftItemsSupplementBackButton = true
+    }
+
+    private func setupWebView() {
+        webView = WKWebView(frame: view.bounds)
+        webView.navigationDelegate = self
+        view.addSubview(webView)
+        view.pinSubviewToAllEdges(webView)
+    }
+
+    @objc private func dismissPreview() {
+        onClose?()
+        presentingViewController?.dismiss(animated: true, completion: nil)
+    }
+
+    @objc private func sharePost() {
+        if let post = post as? Post {
+            let sharingController = PostSharingController()
+            sharingController.sharePost(post, fromBarButtonItem: shareBarButtonItem, inViewController: self)
+        }
+    }
+
+
+    // MARK: Reachability
+
+    private func reloadWhenConnectionRestored() {
+        reachabilityObserver = ReachabilityUtils.observeOnceInternetAvailable { [weak self] in
+            self?.refreshWebView()
+        }
+    }
+
+    private func stopWaitingForConnectionRestored() {
+        if let observer = reachabilityObserver {
+            NotificationCenter.default.removeObserver(observer)
+            reachabilityObserver = nil
+        }
+    }
+
+    // MARK: Loading Animations
+
+    private func startLoadAnimation() {
+        navigationItem.setLeftBarButton(statusButtonItem, animated: true)
+        navigationItem.title = nil
+    }
+
+    private func stopLoadAnimation() {
+        navigationItem.leftBarButtonItem = navigationItem.backBarButtonItem
+        navigationItem.title = NSLocalizedString("Preview", comment: "Post Editor / Preview screen title.")
+    }
+
+    private func showNoResults(withTitle title: String) {
+        let controller = NoResultsViewController.controllerWith(title: title,
+                                                                buttonTitle: NSLocalizedString("Retry", comment: "Button to retry a preview that failed to load"),
+                                                                subtitle: nil,
+                                                                attributedSubtitle: nil,
+                                                                attributedSubtitleConfiguration: nil,
+                                                                image: nil,
+                                                                subtitleImage: nil,
+                                                                accessoryView: nil)
+        controller.delegate = self
+        noResultsViewController = controller
+        addChild(controller)
+        view.addSubview(controller.view)
+        view.pinSubviewToAllEdges(controller.view)
+        noResultsViewController?.didMove(toParent: self)
+    }
+}
+
+extension NewPostPreviewViewController: PostPreviewGeneratorDelegate {
+    func preview(_ generator: PostPreviewGenerator, attemptRequest request: URLRequest) {
+        startLoadAnimation()
+        var newRequest = request
+        newRequest.url = URL(string: "http://sdfsldfjlwefj.sdlfj")
```

Comment:
Is this intentional? The existing implementation just passes the request to the webView.

Reply:
Oops, accidentally pushed this when testing NoResults. 🤦‍♂

Reply:
Oops, accidentally pushed this after testing NoResults 🤦‍♂